lib/Ast_to_Mir.ml

```diff
@@ -100,11 +105,8 @@ let rec trans_stmt {Ast.stmt_typed; stmt_typed_loc; _} =
           targetpe
             (FunApp (distribution.name, List.map ~f:trans_expr (arg :: args)))
         in
-        Block
-          [ Option.value
-              ~default:(bind_loc stmt_typed_loc Skip)
-              (truncate_dist arg truncation)
-          ; bind_loc stmt_typed_loc add_dist ]
+        SList
+          (truncate_dist arg truncation @ [bind_loc stmt_typed_loc add_dist])
     | Ast.Print ps -> NRFunApp ("print", List.map ~f:trans_printable ps)
     | Ast.Reject ps -> NRFunApp ("reject", List.map ~f:trans_printable ps)
     | Ast.IfThenElse (cond, ifb, elseb) ->
```

Comment:
The translation of ForEach loops does not seem right to me.

Reply:
What seems wrong about it?

Reply:
You're translating the foreach loop into a for loop. Say that you're looping over an array of type `int[,]` As it currently stands, the loop variable will change type from `int[]` to `int`. Maybe have a look at what the current code generator does? It also needs to generate different code depending on whether you're looping over an array or over an Eigen type.

Reply:
Also, if you'd want to get rid of language constructs, would it be worth encoding the while loop as a for-loop or vice versa? Not really sure whether that's a good idea. We'll definitely want to identify statically sized for-loops for the loop-unrolling, but not sure if a uniform representation of all loops would get in the way for that.

Reply:
Example we talked about

```
//ORIGINAL stan
real xs[4];
for (x in xs)
  print(x);
  
  
  
  
//generated code (stanc2)
real xs[4];
for (i__ in 1:4) {
  real x = xs[i__];
  print(x);
}



//generated code (stanc3)
real xs[4];
for (x in 1:4) {
  print(x);
}
```